### PR TITLE
feat: Test coverage and automate publish

### PR DESCRIPTION
**Pull request by ankitdas13**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version:  3.1
+    - name: Publish gems to Rubygems
+      run: |
+       mkdir -p $HOME/.gem
+       touch $HOME/.gem/credentials
+       chmod 0600 $HOME/.gem/credentials
+       printf -- "---\n:rubygems_api_key: ${GEM_API_KEY}\n" > $HOME/.gem/credentials
+       gem push gems/*.gem --otp $(oathtool --base32 --totp "$GEM_2FA_SECRET_KEY")
```
Comment (ndhaka007) [inline .github/workflows/ci.yml:77]: Can you explain this or share some resources? What is happening here

Comment (ankitdas13) [reply .github/workflows/ci.yml:77]: Sure, you can check this post 
https://andrewm.codes/blog/automating-ruby-gem-releases-with-github-actions/
and line no 77 i have added 2FA auth, although i am not sure 2fa is mandatory or not in razorpay repo [rubygems](https://rubygems.org/gems/razorpay) until i added as a collaborator. but if there is no 2FA mandatory, then we will remove `--otp $(oathtool --base32 --totp "$GEM_2FA_SECRET_KEY")`

created clone of this repo and publish it (for testing). Check this once 
https://github.com/ankitdas13/razorruby/actions/runs/4805861071

Comment (ndhaka007) [reply .github/workflows/ci.yml:77]: on the shared link it says we are building the image and publishing it. My concern is we have a build step do we need to build again? Can't we directly publish it?

Comment (ankitdas13) [reply .github/workflows/ci.yml:77]: if we want to publish it directly then we have to save the build with help of this `actions/upload-artifact@v3` in git environment and while publish we can use this save build by using this `actions/download-artifact@v3`

**Review by ndhaka007 — APPROVED** [review]
LGTM

**Pull request merged**